lib/src/repo.rs

```diff
@@ -967,3 +967,24 @@ impl MutableRepo {
 #[derive(Debug, Copy, Clone, Error)]
 #[error("Cannot rewrite the root commit")]
 pub struct RewriteRootCommit;
+
+#[derive(Debug, Error)]
+#[error("Cannot access {path}")]
+pub struct PathError {
+    pub path: PathBuf,
+    #[source]
+    pub error: io::Error,
+}
+
+pub(crate) trait IoResultExt<T> {
+    fn path<P: AsRef<Path>>(self, path: P) -> Result<T, PathError>;
```

Comment:
Since `P` is only used once, is `path: impl AsRef<Path>` preferred?

Reply:
I suspect you can't use the `impl` shorthand on `trait` definitions.

Reply:
Isn't `impl` just syntax sugar for the type parameter syntax?

If we needed this in a trait *object*, I think neither form would be allowed.

Reply:
Both do work in this case; I have no particular preference, so went ahead and switched it over.